Code/Framework/AzToolsFramework/Tests/Prefab/PrefabDetachPrefabTests.cpp

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <Prefab/PrefabTestFixture.h>
+
+namespace UnitTest
+{
+    using PrefabDetachPrefabTests = PrefabTestFixture;
+
+    TEST_F(PrefabDetachPrefabTests, DetachPrefabWithEntitySucceeds)
+    {
+        const AZStd::string carPrefabName = "CarPrefab";
+        const AZStd::string tireEntityName = "Tire";
+
+        AZ::IO::Path engineRootPath;
+        m_settingsRegistryInterface->Get(engineRootPath.Native(), AZ::SettingsRegistryMergeUtils::FilePathKey_EngineRootFolder);
+        AZ::IO::Path carPrefabFilepath = engineRootPath / carPrefabName;
+
+        AZ::EntityId tireEntityId = CreateEditorEntityUnderRoot(tireEntityName);
+        AZ::EntityId carContainerId = CreateEditorPrefab(carPrefabFilepath, { tireEntityId });
+
+        InstanceAlias carInstanceAlias = FindNestedInstanceAliasInInstance(GetRootContainerEntityId(), carPrefabName);
+
+        // Detach the car prefab.
+        PrefabOperationResult result = m_prefabPublicInterface->DetachPrefab(carContainerId);
+        ASSERT_TRUE(result.IsSuccess());
+
+        PropagateAllTemplateChanges();
+
+        // Validate there is no nested instance in the level prefab instance.
+        ValidateNestedInstanceNotUnderInstance(GetRootContainerEntityId(), carInstanceAlias);
+
+        InstanceOptionalReference levelInstance = m_instanceEntityMapperInterface->FindOwningInstance(GetRootContainerEntityId());
+        EXPECT_TRUE(levelInstance.has_value());
+
+        // Validate there are two entities in the level prefab instance.
+        EXPECT_EQ(levelInstance->get().GetEntityAliasCount(), 2);
+
+        // Validate the hierarchy is unchanged after detaching.
+        AZStd::string carEntityAliasAfterDetach = FindEntityAliasInInstance(GetRootContainerEntityId(), carPrefabName);
+        AZ::EntityId carEntityIdAfterDetach = levelInstance->get().GetEntityId(carEntityAliasAfterDetach);
+        AZStd::string tireEntityAliasAfterDetach = FindEntityAliasInInstance(GetRootContainerEntityId(), tireEntityName);
+        AZ::EntityId tireEntityIdAfterDetach = levelInstance->get().GetEntityId(tireEntityAliasAfterDetach);
+        EXPECT_TRUE(carEntityIdAfterDetach.IsValid());
+        EXPECT_TRUE(tireEntityIdAfterDetach.IsValid());
+
+        AZ::EntityId parentEntityIdToCheck;
+        AZ::TransformBus::EventResult(parentEntityIdToCheck, tireEntityIdAfterDetach, &AZ::TransformInterface::GetParentId);
+        EXPECT_EQ(carEntityIdAfterDetach, parentEntityIdToCheck);
+    }
+
+    TEST_F(PrefabDetachPrefabTests, DetachPrefabWithNestedPrefabSucceeds)
+    {
+        // Level
+        // | Car       (prefab)  <-- detach prefab
+        //   | Wheel   (prefab)
+        //     | Tire
```

Comment:
Another important use case to test would be to detach a prefab that is parented to another entity within the owning prefab. Example:

- Level
  - entity1
    - car.prefab <-- detach
      - wheel entity